qutip/tests/test_random.py

```diff
@@ -71,13 +72,34 @@ def check_func_dims(func, args, kwargs, dims):
     # TODO: promote this out of test_random, as it's generically useful
     #       in writing tests.
     resdims = func(*args, **kwargs).dims
-    assert_(resdims == dims, "Checking {}; epected dimensions of {}, got {}.".format(func.__name__, dims, resdims))
+    assert_(resdims == dims, "Checking {}; expected dimensions of {}, got {}.".format(func.__name__, dims, resdims))
+
+def check_func_shape(func, args, kwargs, shape):
+    resdims = func(*args, **kwargs).dims
+    assert_(dims_to_tensor_shape(resdims)==shape,"Checking {}; expected shape of {}, got {}.".format(func.__name__, shape, dims_to_tensor_shape(resdims)))
+
+def check_func_N(func, args, kwargs, N):
+    new_state_shape=func(*args, **kwargs).shape
+    assert_(new_state_shape[0]==N,"Checking {}; expected N of {}, got {}.".format(func.__name__, N, new_state_shape[0]))
```

Comment:
If you're going to test the shape (a good idea!) it might be better to check the _entire_ shape, i.e. do
```python
assert new_state_shape == (N, 1)
```
since we know it exactly.